src/test_lib_json/main.cpp

```diff
@@ -307,7 +307,21 @@ JSONTEST_FIXTURE(ValueTest, arrayIssue252) {
   }
   // JSONTEST_ASSERT_EQUAL(5, root["array"].size());
 }
+JSONTEST_FIXTURE(ValueTest, arrayIssue691) {
+  Json::Value array2;
+  array2.append(10);  // index 0
+  array2.append(20);  // index 1
+  array2.append(30);  // index 2
+  array2.append(50);  // index 3
 
+  JSONTEST_ASSERT_EQUAL(Json::Value(50),array2[3]);
+
+  array2.insert(3,40); // index 3
+  // After updating, index 3 should be changed from 50 to 40, and 50 moved to 
+  // index 4.
+  JSONTEST_ASSERT_EQUAL(Json::Value(40),array2[3]);
+  JSONTEST_ASSERT_EQUAL(Json::Value(50),array2[4]);    
```

Comment:
Should also make sure the preceding elements haven't been corrupted by the operation.
It's easy to just check the whole container. Maybe copy the array out into a std::vector and check for == against an expected std::vector value?

Reply:
Uh, I think this is unnecessary, the `operator[ ]` is base on the map, we get the element by specifying the key value, which is unique, not by the index position.

Reply:
I don't follow, but the insert was shuffling and overwriting elements and I would like to check that we didn't overwrite some stray elements during the insert.